python/jiminy_py/src/jiminy_py/simulator.py

```diff
@@ -47,14 +49,91 @@
+                            Optional: True by default.
+    :param config_path: Configuration toml file to import. It will be
+                        imported AFTER loading the hardware description
+                        file. It can be automatically generated from an
+                        instance by calling `export_config_file` method.
+                        Optional: Looking for '\*_options.toml' file in the
+                        same folder and with the same name. If not found,
+                        using default configuration.
+    :param avoid_instable_collisions: Prevent numerical instabilities by
+                                        replacing collision mesh by vertices
+                                        of associated minimal volume bounding
+                                        box, and replacing primitive box by
+                                        its vertices.
+    :param debug: Whether the debug mode must be activated. Doing it
+                    enables temporary files automatic deletion.
+    """
+    # Check if robot name is valid
+    if name != re.sub('[^A-Za-z0-9_]', '_', name):
+        raise Exception("The name of the robot should be case-insensitive "
+                        "ASCII alphanumeric characters plus underscore.")
+
+    # Generate a temporary Hardware Description File if necessary
+    if hardware_path is None:
+        hardware_path = str(pathlib.Path(
+            urdf_path).with_suffix('')) + '_hardware.toml'
+        if not os.path.exists(hardware_path):
+            # Create a file that will be closed (thus deleted) at exit
+            urdf_name = os.path.splitext(os.path.basename(urdf_path))[0]
+            fd, hardware_path = tempfile.mkstemp(
+                prefix=f"{urdf_name}_", suffix="_hardware.toml")
+            os.close(fd)
+
+            if not debug:
+                def remove_file_at_exit(file_path: str) -> None:
+                    try:
+                        os.remove(file_path)
+                    except (PermissionError, FileNotFoundError):
+                        pass
+
+                atexit.register(partial(
+                    remove_file_at_exit, hardware_path))
+
+            # Generate default Hardware Description File
+            generate_default_hardware_description_file(
+                urdf_path, hardware_path, verbose=debug)
+
+    # Build the robot
+    robot = BaseJiminyRobot(name)
+    robot.initialize(
+        urdf_path, hardware_path, mesh_path_dir, (), has_freeflyer,
+        avoid_instable_collisions, load_visual_meshes=debug, verbose=debug)
+
+    return robot
+
 class Simulator:
-    """Single-robot simulation wrapper providing a unified user-API on top of
+    """Simulation wrapper providing a unified user-API on top of
     the low-level jiminy C++ core library and Python-native modules for 3D
-    rendering and log data visualization.
+    rendering and log data visualization. The simulation can now be multi-robot
```

Comment:
Add more details about how it works under the hood (empty robot name by default plus its special meaning...etc). Explain that methods that refer to a single robot are systematically returning info associated with the first robot, no matter if the simulation is single or multi robot... Explain what is not supported by multi-robot... A lot of info are missing here